File: 424.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = defaultdict(lambda:0)
        start_index = 0
        longest = 0
        max_till_now = 0
        for index, letter in enumerate(s):
            count[letter] += 1
            max_till_now = max(max_till_now, count[letter])
            while(index - start_index + 1 - max_till_now) > k:
                count[s[start_index]] -= 1
                start_index += 1
            longest = max(longest, index - start_index + 1)
        return longest

File: 424.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> count;
        int start = 0, longest = 0, current_longest = 0;

        for(int i = 0; i < s.size(); i++) {
            count[s[i]]++;
            if(count[s[i]] > current_longest) {
                current_longest = count[s[i]];
            }
            while ((i - start + 1 - current_longest) > k) {
                count[s[start]]--;
                start++;
            }
            longest = fmax(longest, i - start + 1);
        }
        return longest;
    }
};